File: android/src/main/kotlin/com/jameskarl/amap/map/bean/MapCreationParams.kt
package com.jameskarl.amap.map.bean

import com.jameskarl.amap.map.IJsonEntity

data class MapCreationParams(
        var cameraPosition: CameraPositionData? = null,
        var settings: UiSettings? = null
) : IJsonEntity {
    override fun toString(): String {
        return "MapCreationParams(cameraPositionData=$cameraPosition, settings=$settings)"
    }
}

File: lib/src/amap/map_creation_params.dart
import 'package:amap/amap.dart';

class MapCreationParams {
  CameraPosition cameraPosition;
  UiSettings settings;

  MapCreationParams({
    this.cameraPosition,
    this.settings,
  });

  Map<String, dynamic> toJson() {
    return {
      'cameraPosition': this.cameraPosition,
      'settings': this.settings,
    };
  }

  factory MapCreationParams.fromJson(Map<String, dynamic> map) {
    return new MapCreationParams(
      cameraPosition: CameraPosition.fromJson(map['cameraPosition']),
      settings: UiSettings.fromJson(map['settings']),
    );
  }

  @override
  String toString() {
    return 'MapCreationParams{cameraPosition: $cameraPosition, settings: $settings}';
  }
}
